src/db.js

```diff
@@ -10,33 +10,66 @@ if (flags.debug) {
 
 let db = null;
 
+const setConnect = () => {
+  const connect = mongoose.connect(
+    connectionDBString,
+    { useMongoClient: true },
+    (error) => {
+      if (error) {
+        logger.error(`Database connection can not be created:\n${error}`);
+      }
+    }
+  );
+  return connect;
+};
+
 class MongoConnetor {
-  connect = () => {
+  connect = async () => {
     if (db === null) {
-      db = mongoose.connect(connectionDBString, {
-        useMongoClient: true
+      db = setConnect();
+
+      logger.info('Database connection was created');
+
+      const self = this;
+      db.connection.on('error', (errorConnection) => {
+        logger.error(`Database connection error:\n${errorConnection}`);
+        self.tryReopen();
       });
     } else {
-      throw new Error(
-        'DB instance already exists, use existing connection or close it before creating a new one.'
+      logger.info(
+        'Connection already exists, use "getConnection" for get existing connection or do close it before creating a new one.'
+      );
+    }
+    return db;
+  };
+
+  tryReopen = () => {
+    logger.error('Trying reopen database connection');
+    try {
+      this.close();
+      this.connect();
+    } catch (error) {
+      logger.error(
+        `Cannot reopen connect, app will close with error status:\n${error}`
       );
+      process.exit(1);
     }
   };
 
   getConnection = () => {
     if (db) {
       return db;
     }
-    throw new Error("DB connection doesn't exist yet.");
+    throw new MissConnectionError();
```

Comment:
This is cool, but not very useful. You can have a simple Error here (it's gonna be 500)

Reply:
Oh, I suppose, that I named Error incorrect and this caused confusion. It's throws when developer trying get db instance without setting connection by "connect()". So in this case we didn't have response object and it's not http error, did it?
I will return it to last changes.